src/web/releases.rs

```diff
@@ -233,6 +257,51 @@ async fn get_search_results(
     })
 }
 
+// Categorize  errors from registry
+fn handle_registry_error(err: anyhow::Error) -> SearchError {
+    // Capture crates.io API error
+    if let Some(registry_request_error) = err.downcast_ref::<reqwest::Error>()
+        && let Some(status) = registry_request_error.status()
+        && (status.is_client_error() || status.is_server_error())
+    {
+        return SearchError::CratesIo(format!(
+            "crates.io returned {status}: {registry_request_error}"
+        ));
+    }
+
+    // Errors from bail!() in RegistryApi::search()
```

Comment:
I don't see why we would match strings differentiate errors coming from a function in the same codebase. 

I remember we already had a partial discussion about this. 

I'm open to other approaches, that don't involve string matching